src/Microsoft.CodeAnalysis.Razor/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -136,13 +136,17 @@ private ImmutableArray<IParameterSymbol> GetInvokeMethodParameters(INamedTypeSym
 
             if (methods.Length == 0)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_CannotFindMethod(ViewComponentTypes.SyncMethodName, ViewComponentTypes.AsyncMethodName, componentType.ToDisplayString(FullNameTypeDisplayFormat)));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_CannotFindMethod(
+                    ViewComponentTypes.SyncMethodName,
+                    ViewComponentTypes.AsyncMethodName,
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat)));
             }
             else if (methods.Length > 1)
             {
-                throw new InvalidOperationException(
-                    ViewComponentResources.FormatViewComponent_AmbiguousMethods(componentType.ToDisplayString(FullNameTypeDisplayFormat), ViewComponentTypes.AsyncMethodName, ViewComponentTypes.SyncMethodName));
+                throw new InvalidOperationException(ViewComponentResources.FormatViewComponent_AmbiguousMethods(
+                    componentType.ToDisplayString(FullNameTypeDisplayFormat),
+                    ViewComponentTypes.AsyncMethodName,
+                    ViewComponentTypes.SyncMethodName));
```

Comment:
@NTaylorMullen - do you know what we throw an exception here? This doesn't seem right. This is currently failing some tests but I wanted to understand why this isn't just a diagnostic.